src/agents/context_broker/ContextBrokerProcessor.cc

```
@@ -0,0 +1,286 @@
+    shared_ptr<ContextBrokerProxy> proxy(new ContextBrokerProxy());
+    return proxy;
+}
+
+void ContextBrokerProcessor::run_command(shared_ptr<BusCommandProxy> proxy) {
+    lock_guard<mutex> semaphore(this->query_threads_mutex);
+    auto context_broker_proxy = dynamic_pointer_cast<ContextBrokerProxy>(proxy);
+    string thread_id = "thread<" + proxy->my_id() + "_" + std::to_string(proxy->get_serial()) + ">";
+    LOG_DEBUG("Starting new thread: " << thread_id << " to run command: <" << proxy->get_command()
+                                      << ">");
+    if (this->query_threads.find(thread_id) != this->query_threads.end()) {
+        Utils::error("Invalid thread id: " + thread_id);
+    } else {
+        shared_ptr<StoppableThread> stoppable_thread = make_shared<StoppableThread>(thread_id);
+        stoppable_thread->attach(new thread(&ContextBrokerProcessor::thread_process_one_query,
+                                            this,
+                                            stoppable_thread,
+                                            context_broker_proxy));
+        this->query_threads[thread_id] = stoppable_thread;
+    }
+}
+
+// -------------------------------------------------------------------------------------------------
+// Private methods
+
+void ContextBrokerProcessor::thread_process_one_query(shared_ptr<StoppableThread> monitor,
+                                                      shared_ptr<ContextBrokerProxy> proxy) {
+    try {
+        if (proxy->args.size() < 2) {
+            Utils::error("Syntax error in query command. Missing implicit parameters.");
+        }
+
+        proxy->untokenize(proxy->args);
+
+        // Do not allow use_cache == false and enforce_cache_recreation == true
+        if (!proxy->get_use_cache() && proxy->get_enforce_cache_recreation()) {
+            LOG_ERROR("use_cache == false and enforce_cache_recreation == true is not allowed");
```

Instead of throwing the error directly, call `Utils::error("Error message", false);` The second (optional) parameter can be used to prevent an exception to be raised.

---

But in this case we do want to raise an exception, right? So it would be just `Utils::error("Error message")`?